src/ray/core_worker/transport/direct_task_transport.cc

```diff
@@ -20,25 +20,52 @@ namespace ray {
 
 Status CoreWorkerDirectTaskSubmitter::SubmitTask(TaskSpecification task_spec) {
   RAY_LOG(DEBUG) << "Submit task " << task_spec.TaskId();
+
+  // The creation of actor is divided into two steps:
```

Comment:
This method gets called asynchronously, on the background event loop. You'll have to move the synchronous code into `core_worker.cc`.